libbpf-tools/slabratetop.bpf.c

```diff
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause) */
+/* Copyright (c) 2022 Rong Tao */
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "slabratetop.h"
+#include "stat.h"
+
+#define MAX_ENTRIES	10240
+
+const volatile pid_t target_pid = 0;
+
+static struct slabrate_info slab_zero_value = {};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, char *);
+	__type(value, struct slabrate_info);
+} slab_entries SEC(".maps");
+
+static int probe_entry(struct kmem_cache *cachep)
+{
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u32 pid = pid_tgid >> 32;
+	struct slabrate_info *valuep;
+    const char *name = BPF_CORE_READ(cachep, name);
+
+	if (target_pid && target_pid != pid)
+		return 0;
+
+	valuep = bpf_map_lookup_elem(&slab_entries, &name);
+	if (!valuep) {
+		bpf_map_update_elem(&slab_entries, &name, &slab_zero_value, BPF_ANY);
+		valuep = bpf_map_lookup_elem(&slab_entries, &name);
+		if (!valuep)
+			return 0;
+    	bpf_probe_read_kernel(&valuep->name, sizeof(valuep->name), name);
+	}
+
+	valuep->count ++;
+	valuep->size += BPF_CORE_READ_BITFIELD_PROBED(cachep, size);
```

Comment:
`size` is not a bitfield, no ?

Reply:
> `size` is not a bitfield, no ?

Yep, fix https://github.com/iovisor/bcc/pull/3797/commits/9f184dfd793f427fb5a0006d6d34b17774c6fb11